### PR TITLE
photon-ant Living Guide

### LINKED ISSUE #5: Is there a way to preview the rendering of the theme?
I made some changes to the theme less files but I don't know if the rendering is good enough. Is there a way to try it?

### PR DESCRIPTION
**Pull request by andymikulski**
Fixes #5 

This PR introduces the 'living style guide' for photon-ant. Included:

 - Site rip of [ant.design](https://ant.design/)
   - `photon-ant` source files directly integrate for easy editing - with live reload support!
   - Extraneous content has been removed (irrelevant design docs, etc).
- `.npmignore` file allows the `demo` folder to live in the repo, but is _not_ included when the package is installed via npm/yarn.
- `setup-demo` and `start` scripts to install and run the living guide locally
   - `setup-demo` installs the packages needed for the demo site to run
  - `start` loads the demo site at `localhost:8001`

This is a big PR, not totally sure the best way to reason about reviewing it. Any questions or feedback, please let me know!

### CODE REVIEW

**Comment by Natim** [comment]
Do you think we could deploy this branch to gh-pages?

**Comment by andymikulski** [comment]
Sorry for the delay, @Natim , I've been out sick the past few work days.

I think deploying a copy of the built living guide to `gh-pages` is desirable, as we can then get style guide with the most recent `photon-ant` updates. However, for development, I think it is still desirable to package the living guide into the `photon-ant` directory as is done here. This setup allows speedy development, without the need to mess with multiple branches while working locally.

That said, if there is a clean way to set up the `gh-pages` branch in such a way it works with dev, I'm totally open to that. I don't have a ton of experience working with `gh-pages`, so there's probably a better workflow which I am unaware of.

**Comment by bwinton** [comment]
For Photon, we do something [like this](https://github.com/FirefoxUX/photon/blob/master/deploy.sh) and then call that [from Travis](https://github.com/FirefoxUX/photon/blob/master/.travis.yml) to auto-update the `gh-pages` branch when something changes on master. Maybe something similar here would work?

**Comment by Natim** [comment]
We can help with gh-pages setup since we use it everywhere for the past 2 years.
Basically you have a gh-pages npm module that will do all the work for you.

I guess you can add ``gh-pages`` as a ``devDependency`` and then a ``deploy`` script in your [package.json](https://github.com/Kinto/kinto-admin/blob/master/package.json#L13) that run:

    cd ant-design/demo/ && npm run build && gh-pages --add --dist build/

**Comment by bwinton** [comment]
(That seems easier than the stuff I do. 😃)

**Comment by Natim** [comment]
@bwinton Also it doesn't run automatically in Travis :) I like the fact that it can be automatically redeployed as soon as something is merged into master.

**Comment by Natim** [comment]
I started a pull-request here: https://github.com/andymikulski/photon-themes/pull/1
But I was unable to build the demo.

**Comment by Natim** [comment]
Feel free to try it: http://firefoxux.github.io/photon-themes

**Comment by bwinton** [comment]
I'm not seeing it show up… Does this need to be merged into this branch first?

**Comment by Natim** [comment]
@andymikulski needs to run `npm run deploy` in the ant-design repository first.

**Comment by andymikulski** [comment]
@Natim Running `deploy` sends the `gh-page` business to my [fork](https://github.com/andymikulski/photon-themes/tree/gh-pages). Do I then need to open a new branch/PR to transfer over the updates? Forgive me if this is simple, just want to ensure I don't push something incorrectly (not that there's much to lose at this point, but..)

**Comment by Natim** [comment]
It is up to you, you can either checkout your fork gh-pages branch and push it here:

    git checkout andymikulski/gh-pages
    git push origin gh-pages

Or you can wait until we merge this in master to run it there.
In the meantime we can use https://andymikulski.github.io/photon-themes/ but it doesn't seems to load

**Comment by andymikulski** [comment]
Yeah, I have a bit of redirection for a personal site which messes with my github pages, hence not having a ton of hands on experience using them!

@Natim you mentioned having problems building the demo - could you elaborate on that? I'd rather determine a fix before landing the PR, if possible.

**Comment by Natim** [comment]
Let me try again:

    nvm use 8
    cd ant-design/
    rm -fr node_modules/  demo/node_modules/
    yarn setup-demo
    yarn start  # The browser opens
    
    ERROR  Failed to compile with 25 errors
    These relative modules were not found:
    * ../icon in ./components/alert/index.tsx, ./components/avatar/index.tsx and 22 others
    * ./icon in ./components/index.tsx

**Comment by andymikulski** [comment]
Weird! Thank you for posting the log, I will investigate locally and see what's up.

**Comment by andymikulski** [comment]
@Natim mind giving this another go when you get the chance? Git was giving me trouble tracking the `Icon` component folder, which was the source of your issues. Hopefully that is the only problem which needs resolving before landing this PR.

**Comment by Natim** [comment]
So it compiles :)

However when I try to load it I have a blank page (the same as in your github.io deploy)

http://localhost:8001/index-1.css is 404 as well as http://localhost:8001/index-2.css

capture d ecran de 2018-01-26 10-46-22

**Comment by andymikulski** [comment]
Bummer, alright. I will investigate. So sorry for the inconvenience!

**Comment by Natim** [comment]
No worries, I am just trying to help you debug but I am not blocked on this feature :)

**Comment by andymikulski** [comment]
Alright, everything seems to be in order. The `index-1`/`-2` files you had seen were actually outdated, so I've removed those references entirely. I've tried cloning the branch on its own and it all seems to build correctly, so _hopefully_ this should be all set.

**Comment by Natim** [comment]
I now see: Error: Cannot find module 'ipaddr.js'

**Comment by Natim** [comment]
```
 ERROR  Failed to compile with 1 errors                                                      19:04:35

 error  in ./site/theme/static/index.less

Module build failed: 

@import 'node_modules/photon-colors/photon-colors.less';
^
Can't resolve './node_modules/photon-colors/photon-colors.less' in '/home/rhubscher/mozilla/photon-themes/ant-design'
      in /home/rhubscher/mozilla/photon-themes/ant-design/index.less (line 1, column 0)

 @ ./site/theme/static/index.less 4:14-199 18:2-22:4 19:20-205
 @ ./site/theme/static/style.js
 @ ./site/theme/template/Layout/index.jsx
 @ ./site/theme ^\.\/template.*$
 @ ./node_modules/bisheng/tmp/routes.index.js
 @ ./node_modules/bisheng/tmp/entry.index.js
 @ multi (webpack)-dev-server/client?http://localhost webpack/hot/dev-server ./node_modules/bisheng/tmp/entry.index.js
```

**Comment by andymikulski** [comment]
Is your `photon-colors` up to date? There was an update to `package.json` somewhere in these commits, you may need to run `yarn` to ensure it's up to date.

**Comment by andymikulski** [comment]
I'm going to go ahead and close this PR, but I will keep the fork/branch alive in the event someone is interested in trying to wrap this up!
